src/target/riscv/riscv.c

```diff
@@ -3675,6 +3700,16 @@ static const struct command_registration riscv_exec_command_handlers[] = {
 			"expose. custom0 is accessed as abstract register number 0xc000, "
```

Comment:
Why does it have to be executed before `init`?
Might the GDB user want to do `monitor ...` to change the register list?

Will the commands work via the OpenOCD telnet interface?

See here for my comment about identifying registers by name:

* https://github.com/riscv/riscv-openocd/pull/787#issuecomment-1380934928

Reply:
> Why does it have to be executed before init?
Because list of available registers is built/initialized during target examination procedure.

> Might the GDB user want to do monitor ... to change the register list?
> Will the commands work via the OpenOCD telnet interface?

This won't work. There are a couple of reasons for that. 
1. As is - the command  can be run during the configuration stage only. That is before gdbserver client can connect.  
2. AFAIK the remote target configuration is requested by gdb only once upon connection. Maybe there are ways to re-request it, but I'm not aware of them currently (+ see the next item).
3. The way registers are initialized in OpenOCD complicates things. These are initialized during the examine procedure. And making OpenOCD to re-examine target may imply way too many changes.